src/snowflake/cli/_plugins/nativeapp/codegen/templating/templating_processor.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+
+import jinja2
+from snowflake.cli._plugins.nativeapp.artifacts import BundleMap
+from snowflake.cli._plugins.nativeapp.codegen.artifact_processor import (
+    ArtifactProcessor,
+)
+from snowflake.cli._plugins.nativeapp.exceptions import InvalidTemplateInFileError
+from snowflake.cli.api.cli_global_context import get_cli_context
+from snowflake.cli.api.console import cli_console as cc
+from snowflake.cli.api.project.schemas.native_app.path_mapping import (
+    PathMapping,
+    ProcessorMapping,
+)
+from snowflake.cli.api.rendering.project_definition_templates import (
+    get_client_side_jinja_env,
+)
+from snowflake.cli.api.rendering.sql_templates import (
+    choose_sql_jinja_env_based_on_template_syntax,
+)
+
+
+class TemplatingProcessor(ArtifactProcessor):
+    """
+    Processor class to perform templating on all relevant artifacts (specified in the project definition file).
+    """
+
+    def process(
+        self,
+        artifact_to_process: PathMapping,
+        processor_mapping: Optional[ProcessorMapping],
+        **kwargs,
+    ):
+        """
+        Process the artifact by executing the templating logic on it.
+        """
+        cc.step(f"Processing artifact {artifact_to_process} with templating processor")
```

Comment:
Use a phase for template expansion in general, and a nested step for each file. That's what we do in other processors. We might have to come up with a more robust solution when complex interleavings of processors happen, but it should be sufficient for now at least.

Reply:
`cc.phase` doesn't show up on the screen. If I use `with cc.phase`, I get:
```
Only one phase allowed at a time.
```

Reply:
yup, I forgot the phase is declared in the compiler file.